### PR TITLE
Default Github rate limit hit cache to 1 hour

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
The `github-rate-limit-exceeded-%s` key is being set indefinitely, causing errors to never be enhanced when the rate limit is hit. Default this time to one hour

## How did you test this change?
Test code logic in sandbox

## Are there any deployment considerations?
Monitor rate limits hit

## Does this work require review from our design team?
N/A

### CODE REVIEW

File: backend/redis/utils.go [hunk 1]
```diff
@@ -492,7 +492,12 @@ func (r *Client) GetHubspotCompanies(ctx context.Context, companies interface{})
 }
 
 func (r *Client) SetGithubRateLimitExceeded(ctx context.Context, gitHubRepo string, expirationTime time.Time) error {
-	return r.setFlag(ctx, GithubRateLimitKey(gitHubRepo), true, time.Until(expirationTime))
+	expirationDuration := time.Until(expirationTime)
+	if expirationDuration <= 0 {
```
Comment (Vadman97) [inline backend/redis/utils.go:496]: are we sure that the `expirationTime` was in fact `0` when we were hitting that edge case, or could github have returned some duration that got formatted in a way that resulted in this being a huge number? could we add a unit test that mocks the response from github that was causing that behavior?

**Review by Vadman97 — APPROVED** [review]
🎉

File: backend/redis/utils.go (hunk 1 above)
Comment (SpennyNDaJets) [reply backend/redis/utils.go:496]: Added some tests - I believe this is happening if an invalid time is provided to the Redis cache. This could be nil or a time in the past.

Screenshot 2023-09-19 at 1 57 13 PM
